### PR TITLE
PYTHON-3884 Replace uses of run-test.sh with tox

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

**Comment by blink1073** [comment]
I opened https://jira.mongodb.org/browse/PYTHON-3891 to track the serverless failures.
I put it back in draft because the encryption failures are a regression.

**Comment by blink1073** [comment]
This is good to go now, the windows failure is expected due to cffi not supporting Python 3.12 yet.

File: tox.ini [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 requires =
     tox>=4
+
 envlist =
     # Test using the system Python.
     test,
-    # Test the extra encryption functionality.
-    test-encryption,
+    # Test using the run-tests Evergreen script.
+    test-g,
```
[the remark this answers is not in the document]
Comment (blink1073) [reply tox.ini]: Fixed

**Review by NoahStapp — APPROVED** [review]
LGTM!

**Pull request merged**